### PR TITLE
docs: PolicyMonitor DR

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

`PolicyMonitor` DR

## Why it does that

_Briefly state why the change was necessary._

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Part of #3422 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

**Review by jimmarino — APPROVED** [review]

**Review by wolf4ood — APPROVED** [review]

File: docs/developer/decision-records/2023-09-07-policy-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Generic Properties
+
+## Decision
+
+A `PolicyMonitor` extension component will be implemented.
+
+## Rationale
+
+Ongoing transfers like "streaming" one, where they actually don't get `COMPLETED`, need to be terminated whenever the
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-09-07-policy-monitor/README.md:9]:
```suggestion
Some transfer types, such as streaming never reach the `COMPLETED` state, so they need to be terminated whenever the
```

File: docs/developer/decision-records/2023-09-07-policy-monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+`Policy` on which the parts agreed on is not valid anymore.
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-09-07-policy-monitor/README.md:10]:
```suggestion
`Policy` on which the participants agreed is not valid anymore.
```

File: docs/developer/decision-records/2023-09-07-policy-monitor/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Approach
+
+This component will be implemented using the same model as the state machine we're using for `TransferProcess`:
+
+- The ingestion will be done with a listener on the `TransferProcessStarted` event, that will add an entry to a `PolicyMonitorStore`.
+- The state machine then will continuously fetch the oldest entries leasing them, check the related policy.
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-09-07-policy-monitor/README.md:17]: I would explain the term "leasing", just in a sentence or two

File: docs/developer/decision-records/2023-09-07-policy-monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+- The ingestion will be done with a listener on the `TransferProcessStarted` event, that will add an entry to a `PolicyMonitorStore`.
+- The state machine then will continuously fetch the oldest entries leasing them, check the related policy. 
+  - If the check fails the TP will get completed through a service call (that could be either embedded in the control-plane 
+    or remote through a rest call) and the entry will be removed from the `PolicyMonitorStore`.
+  - otherwise the lease will be broken and the policy be verified again in the next run.
+
+This mechanism will permit to scale the process to multiple instances.
+
+The `PolicyMonitor` will be deployable embedded in the control-plane or separately in a dedicated runtime.
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-09-07-policy-monitor/README.md:24]:
```suggestion
The `PolicyMonitor` will be deployable embedded in the control-plane or separately in a stand-alone runtime.
```

**Review by paullatzelsperger — APPROVED** [review]
LGTM - minor comments aside

**Pull request merged**